### ISSUE TITLE
[BUG] KVM has been disabled by BIOS. KVM must be enabled before creating any Virtual Machines

### ISSUE THREAD

**Issue by lukasmrtvy** [harvester/harvester#3900]
**Describe the bug**

Getting `KVM has been disabled by BIOS. KVM must be enabled before creating any Virtual Machines.` in Host / Basics menu, which is a VM with enabled nested virtualization on Proxmox.

**To Reproduce**
1. Go to 'Host / Basics'

**Expected behavior**

Nested virtualization is working

**Support bundle**

**Environment**
 - Harvester ISO version:  1.1.2
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): Proxmox with Epyc based CPU ( 7H12 )

**Additional context**
```
harvester:/home/rancher # cat /sys/module/kvm_amd/parameters/nested
1
```
```
harvester:/home/rancher # lsmod | grep kvm
kvm_amd               147456  6
ccp                   106496  1 kvm_amd
kvm                  1052672  1 kvm_amd
irqbypass              16384  1 kvm
```
```
harvester:/home/rancher # egrep '(vmx|svm)' --color=always /proc/cpuinfo
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm rep_good nopl cpuid extd_apicid tsc_known_freq pni pclmulqdq ssse3 fma cx16 sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm cmp_legacy svm cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw perfctr_core ssbd ibrs ibpb stibp vmmcall fsgsbase tsc_adjust bmi1 avx2 smep bmi2 rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves clzero xsaveerptr wbnoinvd arat npt nrip_save umip rdpid arch_capabilities
```
```
harvester:/home/rancher # kubectl exec virt-launcher-test-9vp9f -it bash
kubectl exec [POD] [COMMAND] is DEPRECATED and will be removed in a future version. Use kubectl exec [POD] -- [COMMAND] instead.
test:/ # virt-host-validate
  QEMU: Checking for hardware virtualization                                 : PASS
  QEMU: Checking if device /dev/kvm exists                                   : PASS
  QEMU: Checking if device /dev/kvm is accessible                            : PASS
  QEMU: Checking if device /dev/vhost-net exists                             : PASS
  QEMU: Checking if device /dev/net/tun exists                               : PASS
  QEMU: Checking for cgroup 'cpu' controller support                         : PASS
  QEMU: Checking for cgroup 'cpuacct' controller support                     : PASS
  QEMU: Checking for cgroup 'cpuset' controller support                      : PASS
  QEMU: Checking for cgroup 'memory' controller support                      : PASS
  QEMU: Checking for cgroup 'devices' controller support                     : PASS
  QEMU: Checking for cgroup 'blkio' controller support                       : PASS
  QEMU: Checking for device assignment IOMMU support                         : WARN (No ACPI IVRS table found, IOMMU either disabled in BIOS or not supported by this hardware platform)
  QEMU: Checking for secure guest support                                    : WARN (Unknown if this platform has Secure Guest support)
   LXC: Checking for Linux >= 2.6.26                                         : PASS
   LXC: Checking for namespace ipc                                           : PASS
   LXC: Checking for namespace mnt                                           : PASS
   LXC: Checking for namespace pid                                           : PASS
   LXC: Checking for namespace uts                                           : PASS
   LXC: Checking for namespace net                                           : PASS
   LXC: Checking for namespace user                                          : PASS
   LXC: Checking for cgroup 'cpu' controller support                         : PASS
   LXC: Checking for cgroup 'cpuacct' controller support                     : PASS
   LXC: Checking for cgroup 'cpuset' controller support                      : PASS
   LXC: Checking for cgroup 'memory' controller support                      : PASS
   LXC: Checking for cgroup 'devices' controller support                     : PASS
   LXC: Checking for cgroup 'freezer' controller support                     : PASS
   LXC: Checking for cgroup 'blkio' controller support                       : PASS
   LXC: Checking if device /sys/fs/fuse/connections exists                   : PASS
```

**Comment by iosifnicolae2** [comment]
Same problem here with AMD Ryzen 9 5900X

**Comment by w13915984028** [comment]
No matter Harvester is installed on bare metal server, or KVM VM from any linux Host, the user needs to enable `Virtualization` or `Nested Virtualization` manually, Harvester itself CANNOT modify the BIOS directly.

What do you expect Harvester to do in such scenario? thanks.

**Comment by iosifnicolae2** [comment]
I've enabled virtualization from BIOS.

**Comment by w13915984028** [comment]
OK, Let me check it.

cc @bk201 @guangbochen I reproduced in v120 master-head.

<img width="935" alt="image" src="https://github.com/harvester/harvester/assets/31133476/536290ed-3e9e-41dc-b91d-8d1c480122bf">

**Comment by w13915984028** [comment]
Luckily, we can still normally create and run VM, the message seems a bit wired.

<img width="961" alt="image" src="https://github.com/harvester/harvester/assets/31133476/8c06e421-204f-4fa6-8f65-77cfcd499d0e">

**Comment by w13915984028** [comment]
@lukasmrtvy @iosifnicolae2 thanks for your report.

**Comment by w13915984028** [comment]
This seems to be UI message print related.

For UI team to debug:  cc @WuJun2016 @guangbochen @bk201 

<details>
<summary>node description yaml</summary>

```
harv41:~ # kubectl get nodes harv41 -oyaml
apiVersion: v1
kind: Node
metadata:
  annotations:
    cluster.x-k8s.io/cluster-name: local
    cluster.x-k8s.io/cluster-namespace: fleet-local
    cluster.x-k8s.io/machine: custom-9a2a558f853b
    csi.volume.kubernetes.io/nodeid: '{"driver.longhorn.io":"harv41"}'
    etcd.rke2.cattle.io/node-address: 192.168.122.141
    etcd.rke2.cattle.io/node-name: harv41-b8b4e4a0
    flannel.alpha.coreos.com/backend-data: '{"VNI":1,"VtepMAC":"ca:3a:9c:2b:a4:8f"}'
    flannel.alpha.coreos.com/backend-type: vxlan
    flannel.alpha.coreos.com/kube-subnet-manager: "true"
    flannel.alpha.coreos.com/public-ip: 192.168.122.141
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"v1","kind":"Node","metadata":{"annotations":{},"labels":{"node-role.kubernetes.io/etcd":"true"},"name":"harv41"}}
    kubevirt.io/heartbeat: "2023-05-15T13:21:39Z"
    management.cattle.io/pod-limits: '{"cpu":"7","devices.kubevirt.io/kvm":"1","devices.kubevirt.io/tun":"1","devices.kubevirt.io/vhost-net":"1","memory":"9323860Ki"}'
    management.cattle.io/pod-requests: '{"cpu":"3362m","devices.kubevirt.io/kvm":"1","devices.kubevirt.io/tun":"1","devices.kubevirt.io/vhost-net":"1","ephemeral-storage":"50M","memory":"5100884Ki","pods":"66"}'
    node.alpha.kubernetes.io/ttl: "0"
    projectcalico.org/IPv4Address: 192.168.122.141/24
    projectcalico.org/IPv4IPIPTunnelAddr: 10.52.0.1
    rke2.io/encryption-config-hash: start-a7936658947024ee5ef643eb05f3f5b90073a80481e84e37a72aa2548834f0ba
    rke2.io/hostname: harv41
    rke2.io/internal-ip: 192.168.122.141
    rke2.io/node-args: '["server","--node-label","rke.cattle.io/machine=a3d6e840-abcc-4fe2-9e8b-667dd9ba3e57","--token","********","--agent-token","********","--cni","multus,canal","--kube-controller-manager-arg","cert-dir=/var/lib/rancher/rke2/server/tls/kube-controller-manager","--kube-controller-manager-arg","secure-port=10257","--kube-controller-manager-extra-mount","/var/lib/rancher/rke2/server/tls/kube-controller-manager:/var/lib/rancher/rke2/server/tls/kube-controller-manager","--kube-scheduler-arg","cert-dir=/var/lib/rancher/rke2/server/tls/kube-scheduler","--kube-scheduler-arg","secure-port=10259","--kube-scheduler-extra-mount","/var/lib/rancher/rke2/server/tls/kube-scheduler:/var/lib/rancher/rke2/server/tls/kube-scheduler","--cluster-cidr","10.52.0.0/16","--service-cidr","10.53.0.0/16","--cluster-dns","10.53.0.10","--tls-san","192.168.122.144","--audit-policy-file","/etc/rancher/rke2/config.yaml.d/92-harvester-kube-audit-policy.yaml","--apiVersion","audit.k8s.io/v1","--kind","Policy","--omitStages","ResponseStarted","--omitStages","ResponseComplete","--rules","[{level
      Metadata} {verbs [create delete patch]} {omitStages [ResponseStarted ResponseComplete]}]"]'
    rke2.io/node-config-hash: 3WJOIJM65XAEFIAMIPEXQI2YRQ662FRFDDN73TTXYXJMWMQ5I4MQ====
    rke2.io/node-env: '{}'
    volumes.kubernetes.io/controller-managed-attach-detach: "true"
  creationTimestamp: "2023-05-11T12:28:37Z"
  finalizers:
  - wrangler.cattle.io/managed-etcd-controller
  - wrangler.cattle.io/node
  - wrangler.cattle.io/harvester-network-manager-node-controller
  labels:
    beta.kubernetes.io/arch: amd64
    beta.kubernetes.io/instance-type: rke2
    beta.kubernetes.io/os: linux
    cpu-feature.node.kubevirt.io/3dnowprefetch: "true"
    cpu-feature.node.kubevirt.io/abm: "true"
    cpu-feature.node.kubevirt.io/adx: "true"
    cpu-feature.node.kubevirt.io/aes: "true"
    cpu-feature.node.kubevirt.io/amd-ssbd: "true"
    cpu-feature.node.kubevirt.io/amd-stibp: "true"
    cpu-feature.node.kubevirt.io/arat: "true"
    cpu-feature.node.kubevirt.io/arch-capabilities: "true"
    cpu-feature.node.kubevirt.io/avx: "true"
    cpu-feature.node.kubevirt.io/avx2: "true"
    cpu-feature.node.kubevirt.io/bmi1: "true"
    cpu-feature.node.kubevirt.io/bmi2: "true"
    cpu-feature.node.kubevirt.io/clflushopt: "true"
    cpu-feature.node.kubevirt.io/clwb: "true"
    cpu-feature.node.kubevirt.io/clzero: "true"
    cpu-feature.node.kubevirt.io/cmp_legacy: "true"
    cpu-feature.node.kubevirt.io/cr8legacy: "true"
    cpu-feature.node.kubevirt.io/erms: "true"
    cpu-feature.node.kubevirt.io/f16c: "true"
    cpu-feature.node.kubevirt.io/fma: "true"
    cpu-feature.node.kubevirt.io/fsgsbase: "true"
    cpu-feature.node.kubevirt.io/fsrm: "true"
    cpu-feature.node.kubevirt.io/fxsr_opt: "true"
    cpu-feature.node.kubevirt.io/hypervisor: "true"
    cpu-feature.node.kubevirt.io/ibpb: "true"
    cpu-feature.node.kubevirt.io/ibrs: "true"
    cpu-feature.node.kubevirt.io/invpcid: "true"
    cpu-feature.node.kubevirt.io/mds-no: "true"
    cpu-feature.node.kubevirt.io/misalignsse: "true"
    cpu-feature.node.kubevirt.io/mmxext: "true"
    cpu-feature.node.kubevirt.io/movbe: "true"
    cpu-feature.node.kubevirt.io/npt: "true"
    cpu-feature.node.kubevirt.io/nrip-save: "true"
    cpu-feature.node.kubevirt.io/osvw: "true"
    cpu-feature.node.kubevirt.io/pclmuldq: "true"
    cpu-feature.node.kubevirt.io/pdpe1gb: "true"
    cpu-feature.node.kubevirt.io/perfctr_core: "true"
    cpu-feature.node.kubevirt.io/pku: "true"
    cpu-feature.node.kubevirt.io/popcnt: "true"
    cpu-feature.node.kubevirt.io/pschange-mc-no: "true"
    cpu-feature.node.kubevirt.io/rdctl-no: "true"
    cpu-feature.node.kubevirt.io/rdpid: "true"
    cpu-feature.node.kubevirt.io/rdrand: "true"
    cpu-feature.node.kubevirt.io/rdseed: "true"
    cpu-feature.node.kubevirt.io/rdtscp: "true"
    cpu-feature.node.kubevirt.io/sha-ni: "true"
    cpu-feature.node.kubevirt.io/skip-l1dfl-vmentry: "true"
    cpu-feature.node.kubevirt.io/smap: "true"
    cpu-feature.node.kubevirt.io/smep: "true"
    cpu-feature.node.kubevirt.io/spec-ctrl: "true"
    cpu-feature.node.kubevirt.io/ssbd: "true"
    cpu-feature.node.kubevirt.io/sse4.2: "true"
    cpu-feature.node.kubevirt.io/sse4a: "true"
    cpu-feature.node.kubevirt.io/stibp: "true"
    cpu-feature.node.kubevirt.io/svm: "true"
    cpu-feature.node.kubevirt.io/svme-addr-chk: "true"
    cpu-feature.node.kubevirt.io/tsc-deadline: "true"
    cpu-feature.node.kubevirt.io/tsc_adjust: "true"
    cpu-feature.node.kubevirt.io/umip: "true"
    cpu-feature.node.kubevirt.io/vaes: "true"
    cpu-feature.node.kubevirt.io/virt-ssbd: "true"
    cpu-feature.node.kubevirt.io/vme: "true"
    cpu-feature.node.kubevirt.io/vpclmulqdq: "true"
    cpu-feature.node.kubevirt.io/wbnoinvd: "true"
    cpu-feature.node.kubevirt.io/x2apic: "true"
    cpu-feature.node.kubevirt.io/xgetbv1: "true"
    cpu-feature.node.kubevirt.io/xsave: "true"
    cpu-feature.node.kubevirt.io/xsavec: "true"
    cpu-feature.node.kubevirt.io/xsaveerptr: "true"
    cpu-feature.node.kubevirt.io/xsaveopt: "true"
    cpu-feature.node.kubevirt.io/xsaves: "true"
    cpu-model-migration.node.kubevirt.io/Dhyana: "true"
    cpu-model-migration.node.kubevirt.io/EPYC: "true"
    cpu-model-migration.node.kubevirt.io/EPYC-IBPB: "true"
    cpu-model-migration.node.kubevirt.io/EPYC-Milan: "true"
    cpu-model-migration.node.kubevirt.io/EPYC-Rome: "true"
    cpu-model-migration.node.kubevirt.io/IvyBridge: "true"
    cpu-model-migration.node.kubevirt.io/IvyBridge-IBRS: "true"
    cpu-model-migration.node.kubevirt.io/Nehalem: "true"
    cpu-model-migration.node.kubevirt.io/Nehalem-IBRS: "true"
    cpu-model-migration.node.kubevirt.io/Opteron_G1: "true"
    cpu-model-migration.node.kubevirt.io/Opteron_G2: "true"
    cpu-model-migration.node.kubevirt.io/Opteron_G3: "true"
    cpu-model-migration.node.kubevirt.io/Penryn: "true"
    cpu-model-migration.node.kubevirt.io/SandyBridge: "true"
    cpu-model-migration.node.kubevirt.io/SandyBridge-IBRS: "true"
    cpu-model-migration.node.kubevirt.io/Westmere: "true"
    cpu-model-migration.node.kubevirt.io/Westmere-IBRS: "true"
    cpu-model.node.kubevirt.io/Dhyana: "true"
    cpu-model.node.kubevirt.io/EPYC: "true"
    cpu-model.node.kubevirt.io/EPYC-IBPB: "true"
    cpu-model.node.kubevirt.io/EPYC-Rome: "true"
    cpu-model.node.kubevirt.io/IvyBridge: "true"
    cpu-model.node.kubevirt.io/IvyBridge-IBRS: "true"
    cpu-model.node.kubevirt.io/Nehalem: "true"
    cpu-model.node.kubevirt.io/Nehalem-IBRS: "true"
    cpu-model.node.kubevirt.io/Opteron_G1: "true"
    cpu-model.node.kubevirt.io/Opteron_G2: "true"
    cpu-model.node.kubevirt.io/Opteron_G3: "true"
    cpu-model.node.kubevirt.io/Penryn: "true"
    cpu-model.node.kubevirt.io/SandyBridge: "true"
    cpu-model.node.kubevirt.io/SandyBridge-IBRS: "true"
    cpu-model.node.kubevirt.io/Westmere: "true"
    cpu-model.node.kubevirt.io/Westmere-IBRS: "true"
    cpu-vendor.node.kubevirt.io/AMD: "true"
    cpumanager: "false"
    harvesterhci.io/managed: "true"
    host-model-cpu.node.kubevirt.io/EPYC-Milan: "true"
    host-model-required-features.node.kubevirt.io/arch-capabilities: "true"
    host-model-required-features.node.kubevirt.io/cmp_legacy: "true"
    host-model-required-features.node.kubevirt.io/hypervisor: "true"
    host-model-required-features.node.kubevirt.io/mds-no: "true"
    host-model-required-features.node.kubevirt.io/pschange-mc-no: "true"
    host-model-required-features.node.kubevirt.io/rdctl-no: "true"
    host-model-required-features.node.kubevirt.io/skip-l1dfl-vmentry: "true"
    host-model-required-features.node.kubevirt.io/spec-ctrl: "true"
    host-model-required-features.node.kubevirt.io/ssbd: "true"
    host-model-required-features.node.kubevirt.io/stibp: "true"
    host-model-required-features.node.kubevirt.io/tsc-deadline: "true"
    host-model-required-features.node.kubevirt.io/tsc_adjust: "true"
    host-model-required-features.node.kubevirt.io/vaes: "true"
    host-model-required-features.node.kubevirt.io/virt-ssbd: "true"
    host-model-required-features.node.kubevirt.io/vpclmulqdq: "true"
    host-model-required-features.node.kubevirt.io/x2apic: "true"
    hyperv.node.kubevirt.io/base: "true"
    hyperv.node.kubevirt.io/frequencies: "true"
    hyperv.node.kubevirt.io/ipi: "true"
    hyperv.node.kubevirt.io/reenlightenment: "true"
    hyperv.node.kubevirt.io/reset: "true"
    hyperv.node.kubevirt.io/runtime: "true"
    hyperv.node.kubevirt.io/synic: "true"
    hyperv.node.kubevirt.io/synic2: "true"
    hyperv.node.kubevirt.io/synictimer: "true"
    hyperv.node.kubevirt.io/time: "true"
    hyperv.node.kubevirt.io/tlbflush: "true"
    hyperv.node.kubevirt.io/vpindex: "true"
    kubernetes.io/arch: amd64
    kubernetes.io/hostname: harv41
    kubernetes.io/os: linux
    kubevirt.io/schedulable: "true"
    network.harvesterhci.io/mgmt: "true"
    node-role.kubernetes.io/control-plane: "true"
    node-role.kubernetes.io/etcd: "true"
    node-role.kubernetes.io/master: "true"
    node-role.kubernetes.io/worker: "true"
    node.kubernetes.io/instance-type: rke2
    plan.upgrade.cattle.io/sync-additional-ca: 2c27af4f7e139d7c15346edc0adb200edade07eef796ef2c41ef923b
    plan.upgrade.cattle.io/sync-containerd-registry: 425425568e817c6e4c282b875a1677e0ca937ace69c8a7ebd2f2b80d
    plan.upgrade.cattle.io/system-agent-upgrader: 47855a995ec58be3dbfafc42f69cf422dd750c2dbf22d8f6c1e9e446
    rke.cattle.io/machine: a3d6e840-abcc-4fe2-9e8b-667dd9ba3e57
  name: harv41
  resourceVersion: "56145"
  uid: 8648f0dd-8292-470b-90e3-64a0b7d38c99
spec:
  podCIDR: 10.52.0.0/24
  podCIDRs:
  - 10.52.0.0/24
  providerID: rke2://harv41
status:
  addresses:
  - address: 192.168.122.141
    type: InternalIP
  - address: harv41
    type: Hostname
  allocatable:
    cpu: "10"
    devices.kubevirt.io/kvm: 1k
    devices.kubevirt.io/tun: 1k
    devices.kubevirt.io/vhost-net: 1k
    ephemeral-storage: "59935978449"
    hugepages-1Gi: "0"
    hugepages-2Mi: "0"
    memory: 20476496Ki
    pods: "110"
  capacity:
    cpu: "10"
    devices.kubevirt.io/kvm: 1k
    devices.kubevirt.io/tun: 1k
    devices.kubevirt.io/vhost-net: 1k
    ephemeral-storage: 61611820Ki
    hugepages-1Gi: "0"
    hugepages-2Mi: "0"
    memory: 20476496Ki
    pods: "110"
  conditions:
  - lastHeartbeatTime: "2023-05-15T12:52:33Z"
    lastTransitionTime: "2023-05-15T12:52:33Z"
    message: Flannel is running on this node
    reason: FlannelIsUp
    status: "False"
    type: NetworkUnavailable
  - lastHeartbeatTime: "2023-05-15T13:18:23Z"
    lastTransitionTime: "2023-05-11T12:28:36Z"
    message: kubelet has sufficient memory available
    reason: KubeletHasSufficientMemory
    status: "False"
    type: MemoryPressure
  - lastHeartbeatTime: "2023-05-15T13:18:23Z"
    lastTransitionTime: "2023-05-11T12:28:36Z"
    message: kubelet has no disk pressure
    reason: KubeletHasNoDiskPressure
    status: "False"
    type: DiskPressure
  - lastHeartbeatTime: "2023-05-15T13:18:23Z"
    lastTransitionTime: "2023-05-11T12:28:36Z"
    message: kubelet has sufficient PID available
    reason: KubeletHasSufficientPID
    status: "False"
    type: PIDPressure
  - lastHeartbeatTime: "2023-05-15T13:18:23Z"
    lastTransitionTime: "2023-05-11T12:31:15Z"
    message: kubelet is posting ready status. AppArmor enabled
    reason: KubeletReady
    status: "True"
    type: Ready
  daemonEndpoints:
    kubeletEndpoint:
      Port: 10250
  images:
  - names:
    - docker.io/rancher/rancher:v2.7.3
    sizeBytes: 1697707232
  - names:
    - docker.io/rancher/nginx-ingress-controller:nginx-1.4.1-hardened2
    sizeBytes: 799758351
  - names:
    - docker.io/longhornio/longhorn-instance-manager:v1.4.1
    sizeBytes: 746537855
  - names:
    - docker.io/longhornio/longhorn-engine:v1.4.1
    sizeBytes: 744781110
  - names:
    - docker.io/rancher/hardened-kubernetes:v1.24.11-rke2r1-build20230228
    sizeBytes: 679302913
  - names:
    - registry.suse.com/suse/sles/15.4/libguestfs-tools:0.54.0-150400.3.10.4
    sizeBytes: 625865455
  - names:
    - docker.io/rancher/hardened-calico:v3.25.0-build20230209
    sizeBytes: 594606389
  - names:
    - registry.suse.com/suse/sles/15.4/virt-launcher:0.54.0-150400.3.10.4
    sizeBytes: 574996300
  - names:
    - docker.io/rancher/hardened-multus-cni:v3.9.3-build20230109
    sizeBytes: 419024271
  - names:
    - docker.io/longhornio/support-bundle-kit:v0.0.19
    sizeBytes: 416939751
  - names:
    - docker.io/rancher/harvester-upgrade:398d492b
    - docker.io/rancher/harvester-upgrade:master-398d492b-head
    - docker.io/rancher/harvester-upgrade:master-head
    sizeBytes: 415760644
  - names:
    - docker.io/rancher/support-bundle-kit:master-head
    sizeBytes: 414438629
  - names:
    - docker.io/rancher/shell:v0.1.19
    sizeBytes: 352306602
  - names:
    - docker.io/longhornio/backing-image-manager:v1.4.1
    sizeBytes: 351020493
  - names:
    - registry.suse.com/suse/sles/15.4/virt-handler:0.54.0-150400.3.10.4
    sizeBytes: 330743961
  - names:
    - docker.io/rancher/fleet-agent:v0.6.0
    sizeBytes: 328369819
  - names:
    - docker.io/rancher/harvester:master-head
    sizeBytes: 321842002
  - names:
    - docker.io/rancher/rke2-runtime:v1.24.11-rke2r1
    sizeBytes: 308097401
  - names:
    - docker.io/rancher/mirrored-grafana-grafana:9.1.5
    sizeBytes: 305920355
  - names:
    - docker.io/longhornio/longhorn-manager:v1.4.1
    sizeBytes: 280301334
  - names:
    - docker.io/rancher/gitjob:v0.1.37
    sizeBytes: 254614226
  - names:
    - docker.io/rancher/harvester-vm-import-controller:v0.1.4
    sizeBytes: 254450093
  - names:
    - docker.io/rancher/klipper-helm:v0.7.4-build20221121
    sizeBytes: 251528120
  - names:
    - docker.io/rancher/harvester-pcidevices:v0.2.4
    sizeBytes: 239758953
  - names:
    - docker.io/rancher/harvester-webhook:master-head
    sizeBytes: 226435819
  - names:
    - docker.io/rancher/mirrored-banzaicloud-fluentd:v1.14.6-alpine-5
    sizeBytes: 221240054
  - names:
    - docker.io/rancher/hardened-flannel:v0.21.1-build20230208
    sizeBytes: 218007336
  - names:
    - docker.io/rancher/mirrored-prometheus-prometheus:v2.38.0
    sizeBytes: 216548299
  - names:
    - docker.io/longhornio/longhorn-ui:v1.4.1
    sizeBytes: 207416763
  - names:
    - docker.io/longhornio/longhorn-share-manager:v1.4.1
    sizeBytes: 200228964
  - names:
    - registry.suse.com/suse/sles/15.4/virt-controller:0.54.0-150400.3.10.4
    sizeBytes: 191285038
  - names:
    - registry.suse.com/suse/sles/15.4/virt-api:0.54.0-150400.3.10.4
    sizeBytes: 190680203
  - names:
    - docker.io/rancher/hardened-cni-plugins:v1.0.1-build20221011
    sizeBytes: 190301150
  - names:
    - registry.suse.com/suse/sles/15.4/virt-operator:0.54.0-150400.3.10.4
    sizeBytes: 190213872
  - names:
    - docker.io/rancher/harvester-node-disk-manager:master-head
    sizeBytes: 188271087
  - names:
    - docker.io/rancher/harvester-cluster-repo:master
    sizeBytes: 186568835
  - names:
    - docker.io/rancher/harvester-network-controller:master-head
    sizeBytes: 183123517
  - names:
    - docker.io/rancher/hardened-dns-node-cache:1.21.2-build20221011
    sizeBytes: 181523896
  - names:
    - docker.io/rancher/fleet:v0.6.0
    sizeBytes: 180977556
  - names:
    - docker.io/rancher/hardened-coredns:v1.9.3-build20221011
    sizeBytes: 177230042
  - names:
    - docker.io/rancher/rke2-cloud-provider:v1.26.1-build20230210
    sizeBytes: 174089311
  - names:
    - docker.io/rancher/hardened-k8s-metrics-server:v0.6.2-build20221202
    sizeBytes: 173963979
  - names:
    - docker.io/rancher/hardened-etcd:v3.5.4-k3s1-build20221011
    sizeBytes: 170153416
  - names:
    - docker.io/rancher/harvester-node-manager:master-head
    sizeBytes: 158685581
  - names:
    - docker.io/rancher/shell:v0.1.8
    sizeBytes: 150615020
  - names:
    - docker.io/rancher/hardened-cluster-autoscaler:v1.8.5-build20221011
    sizeBytes: 147856186
  - names:
    - registry.suse.com/bci/bci-base:15.4
    sizeBytes: 122656812
  - names:
    - ghcr.io/k8snetworkplumbingwg/whereabouts:v0.6.1-amd64
    sizeBytes: 102273112
  - names:
    - docker.io/rancher/harvester-network-webhook:master-head
    sizeBytes: 98461254
  - names:
    - docker.io/rancher/harvester-network-helper:master-head
    sizeBytes: 94176837
  nodeInfo:
    architecture: amd64
    bootID: 09cc2e7a-39a1-4c9a-97ef-98f7c65feb48
    containerRuntimeVersion: containerd://1.6.15-k3s1
    kernelVersion: 5.14.21-150400.24.60-default
    kubeProxyVersion: v1.24.11+rke2r1
    kubeletVersion: v1.24.11+rke2r1
    machineID: 2532b367c953f58b5f2b7c8e645cdf44
    operatingSystem: linux
    osImage: Harvester master
    systemUUID: 0a9bb9a9-340b-4413-b75e-b1bd2df11d72
  volumesAttached:
  - devicePath: ""
    name: kubernetes.io/csi/driver.longhorn.io^pvc-6b04e1a2-fa4e-46be-8074-016f04c56732
  volumesInUse:
  - kubernetes.io/csi/driver.longhorn.io^pvc-6b04e1a2-fa4e-46be-8074-016f04c5673
 ```
</details>

**Comment by w13915984028** [comment]
@lukasmrtvy @iosifnicolae2 

It is confirmed that the error message is from UI, the VM life cycle management feature is not affected, please skip this message.  A fix is on-going. thanks.

**Comment by noahgildersleeve** [comment]
Tested with v1.1.2 and ipxe-examples VMs. I was able to repro this issue.
![Screenshot_20230515_102802](https://github.com/harvester/harvester/assets/83787952/9dd085a5-08af-4a94-9210-45073cc39bf8)
![Screenshot_20230515_102754](https://github.com/harvester/harvester/assets/83787952/57b084a7-b66b-4d9c-8320-417be6e78667)
[supportbundle_ea3daa86-1848-4fe1-a914-b06cb4fa43e7_2023-05-15T17-28-16Z.zip](https://github.com/harvester/harvester/files/11480406/supportbundle_ea3daa86-1848-4fe1-a914-b06cb4fa43e7_2023-05-15T17-28-16Z.zip)

**Comment by bk201** [comment]
> Same problem here with AMD Ryzen 9 5900X

@iosifnicolae2 I'm so sorry to reiterate. Did you see the issue within VMs? That is, you use VM to run Harvester and try to run more VMs in a nested virtualization manner.
Or you see this image by installing Harvester to a bare-metal machine. Thanks.

**Comment by bk201** [comment]
@Vicente-Cheng Please help check the correct conditions for both AMD and Intel machines. https://github.com/harvester/harvester/issues/2153#issuecomment-1361340229 is just for AMD and true means hardware virtualization is supported. Probably need a combination to detect if flags are presented.

**Comment by Vicente-Cheng** [comment]
> @Vicente-Cheng Please help check the correct conditions for both AMD and Intel machines. [#2153 (comment)](https://github.com/harvester/harvester/issues/2153#issuecomment-1361340229) is just for AMD and true means hardware virtualization is supported. Probably need a combination to detect if flags are presented.

For Intel CPU, we need to check `cpu-feature.node.kubevirt.io/vmx  = true`.
For AMD CPU, we need to check `cpu-feature.node.kubevirt.io/svm  = true`.

So combine that if any flag above is `false`, we could show the warning message.
like this `(VMX_ENABLE == false || SVM_ENABLE == false)` -> show warning message.

BTW, the label may not exist, e.g. when using AMD CPU, the `vmx` label does not exist.

**Comment by iosifnicolae2** [comment]
> > Same problem here with AMD Ryzen 9 5900X
> 
> @iosifnicolae2 I'm so sorry to reiterate. Did you see the issue within VMs? That is, you use VM to run Harvester and try to run more VMs in a nested virtualization manner.
> Or you see this image by installing Harvester to a bare-metal machine. Thanks.
> 

I saw the issue when installing Harvester on a bare-metal machine.

**Comment by richevanscybermyte** [comment]
I am +1 on this. AMD SVM is enabled in the BIOS but harvester shows KVM disabled.
Harvester 1.1.2
AMD CPU 

This does not show up on my Intel CPU systems.

**Comment by noahgildersleeve** [comment]
Tested in v1.2.0-rc2. Not showing error message anymore. It's showing a different error, but I'll create a different bug about that.

![Screenshot_20230620_184941](https://github.com/harvester/harvester/assets/83787952/4d17f8a2-4c73-4c07-8da6-40505f8d78be)

**Comment by WuJun2016** [comment]
> Tested in v1.2.0-rc2. Not showing error message anymore. It's showing a different error, but I'll create a different bug about that.
> 
> ![Screenshot_20230620_184941](https://user-images.githubusercontent.com/83787952/247374019-4d17f8a2-4c73-4c07-8da6-40505f8d78be.png)

@noahgildersleeve  We can track the error in this issue：  https://github.com/harvester/harvester/issues/4103

**Comment by hunghvu** [comment]
Might be related. My host also shows this error, but I can deploy the virtual machine normally.

![image](https://github.com/harvester/harvester/assets/62678592/d5f68c66-155c-430e-b309-fcae14bc808e)
![image](https://github.com/harvester/harvester/assets/62678592/55e1cbde-d907-4685-9ad8-3a6fb59da834)

However, when I used TrueNAS Scale before, it couldn't detect nested virtualization (test VM created on bare metal TrueNAS before migrating to Harvester) and essentially made the TrueNAS Scale VM inoperable. Therefore, I made a clean TrueNAS Core install without any VM and it has been fine. I will see if this is the case when v1.2.0 is released.

![image](https://github.com/harvester/harvester/assets/62678592/080b8c69-7cf9-4b7f-ba14-e04ce0757d80)
